packages/commonwealth/server/passport/magic.ts

```diff
@@ -454,11 +459,12 @@ async function magicLoginRoute(
         throw new Error('sessionPayload signed with invalid signature');
       }
     }
+
     if (chainToJoin) {
-      if (
-        chainToJoin.base === ChainBase.CosmosSDK &&
-        session.payload.chain.startsWith('cosmos:')
-      ) {
+      if (chainToJoin.base === ChainBase.CosmosSDK) {
+        // throws if magicAddress does not match signed address in didToken
+        await magic.token.validate(req.body.didToken, req.body.magicAddress);
```

Comment:
Reading the code for magic.token.validate -- looks like they set a 5 minute expiration. Question is if we feel confident about that level of security guarantee, knowing that someone could perform a replay attack within that timespan, since the app doesn't know whether a particular token has been consumed.

Reply:
Yeah, we can set the lifetime lower too. Looking into using this in the session signature to utilize that nonce...

Reply:
Now rejecting tokens that have been used before. No longer adding a second magic DID token because we can get the Cosmos address from the bearer (same thing).

Per magic, every DID token has a unique TID, so we can enforce unique sign-ins.